cpp/test/neighbors/ann_cagra.cuh

```diff
@@ -1006,21 +1014,22 @@ inline std::vector<AnnCagraInputs> generate_inputs()
                                                    {1.0f, 2.0f, 3.0f});
   inputs.insert(inputs.end(), inputs2.begin(), inputs2.end());
 
-  inputs2 = raft::util::itertools::product<AnnCagraInputs>(
-    {100},
-    {1000},
-    {1, 3, 5, 7, 8, 17, 64, 128, 137, 192, 256, 512, 619, 1024},  // dim
-    {10},
-    {graph_build_algo::IVF_PQ},
-    {search_algo::AUTO},
-    {10},
-    {0},  // team_size
-    {64},
-    {1},
-    {cuvs::distance::DistanceType::L2Expanded},
-    {false},
-    {false},
-    {0.995});
+  // Varying dim, adding non_owning_memory_buffer_flag
+  inputs2 =
+    raft::util::itertools::product<AnnCagraInputs>({100},
+                                                   {1000},
+                                                   {1, 5, 8, 64, 137, 256, 619, 1024},  // dim
```

Comment:
Here we test a few edge cases where the dimensionality is `2ⁿ` or `2ⁿ ± 1`. These are important to check whether we have any problems with padding the data during build and search.

Reply:
I figured that was the case. There were a couple places that varied `dim`. I kept different values in each test so that the full range would be covered. Perhaps we can cover the full range in only one set of inputs instead, and use a small range for the other cases?